### PR TITLE
SOF-6487: Replace docker images in materials-designer CI with prebuilt

### PR DESCRIPTION
**Pull request by unsigned6**

### CODE REVIEW

File: tests/cucumber/support/widgets/threejs_editor_widget.js [hunk 1]
```diff
@@ -100,9 +100,11 @@ export class ThreeJSEditorWidget extends Widget {
...
                 () => {
                     exabrowser.scrollAndClick(selector);
-                    exabrowser.keys(value);
-                    exabrowser.keys("Enter");
-                    assert(exabrowser.getValue(selector), value);
+                    exabrowser.setValueWithBackspaceClear(selector, value);
+                    // exabrowser.scrollAndClick(this._selectors.wrapper);
+                    // exabrowser.keys(value);
+                    // exabrowser.keys("Enter");
+                    // assert(exabrowser.getValue(selector), value);
```
Comment (timurbazhirov) [inline tests/cucumber/support/widgets/threejs_editor_widget.js]: Remove commented out code, pls.

**Review by timurbazhirov — APPROVED** [review]

File: .github/workflows/cicd.yml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 name: Continuous Testing and Automated NPM Publication
 
-on: [push]
+on: [ push ]
```
Comment (timurbazhirov) [inline .github/workflows/cicd.yml]: Why the spaces??

Comment (unsigned6) [reply .github/workflows/cicd.yml]: looks like autofixed by editor

File: tests/cucumber/support/widgets/threejs_editor_widget.js [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import assert from "assert";
+// import assert from "assert";
```
Comment (timurbazhirov) [inline tests/cucumber/support/widgets/threejs_editor_widget.js]: Remove

File: dockerfiles/app/Dockerfile [hunk 4]
```diff
@@ -1,4 +1,4 @@
-ARG BASE_IMAGE
+ARG BASE_IMAGE=ghcr.io/exabyte-io/centos-7.9-base
```
Comment (timurbazhirov) [inline dockerfiles/app/Dockerfile]: Should we try `https://hub.docker.com/_/node`? 14-alpine

File: entrypoint.sh [hunk 5]
```diff
@@ -10,14 +10,16 @@ fi
...
 export DEBUG="exachimp:*"
 export ROOT_URL="http://127.0.0.1:3001"
+export DISPLAY=":99"
 
 # Chimp can also be started within an xvfb-run command
 # by prepending the following line to the chimp call.
 # xvfb-run -s "-ac -screen 0 1920x1080x24"
 
 if [[ "$1" == "test" ]]; then
+    Xvfb -ac :99 -screen 0 1280x1024x16 &
```
Comment (timurbazhirov) [inline entrypoint.sh]: $DISPLAY

File: dockerfiles/app/Dockerfile (hunk 4 above)
Comment (unsigned6) [reply dockerfiles/app/Dockerfile]: Btw, I remeber that thish centos image comes from this Dockerfile: https://github.com/Exabyte-io/materials-designer/blob/dev/dockerfiles/centos/Dockerfile

File: .github/workflows/cicd.yml [hunk 6]
```diff
@@ -25,26 +25,33 @@ jobs:
...
+      - name: Validate JS package
+        uses: ./actions/js/validate
         with:
-          dockerfile: ./dockerfiles/app/test/Dockerfile
+          node-version: 12.21.x
+
+      #- name: Validate test dockerfile
+      #  uses: ./actions/docker/validate
+      #  with:
+      #    dockerfile: ./dockerfiles/app/test/Dockerfile
```
Comment (timurbazhirov) [inline .github/workflows/cicd.yml]: We should re-enable validation

File: dockerfiles/app/Dockerfile [hunk 7]
```diff
@@ -1,15 +1,17 @@
...
+ARG BASE_IMAGE=node
+ARG IMAGE_TAG=14-alpine
 FROM ${BASE_IMAGE}:${IMAGE_TAG}
 
 WORKDIR /opt/app
 
-COPY . .
+# use changes to package.json to force Docker not to use the cache
+# when we change our application's nodejs dependencies:
+COPY package.json package.json
```
Comment (timurbazhirov) [inline dockerfiles/app/Dockerfile:9]: Add https://stackoverflow.com/questions/35774714/how-to-cache-the-run-npm-install-instruction-when-docker-build-a-dockerfile

File: .github/workflows/cicd.yml [hunk 8]
```diff
@@ -10,13 +10,13 @@ concurrency:
 jobs:
 
   run-tests:
-    runs-on: ubuntu-20.04
+    runs-on: [ubuntu-20.04, self-hosted]
```
Comment (timurbazhirov) [inline .github/workflows/cicd.yml]: Should be one or the other

**Review by timurbazhirov — APPROVED** [review]

**Pull request merged**